components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/obsrvability/handler/metrics/publisher/prometheus/reporter/PrometheusReporter.java

```diff
@@ -0,0 +1,266 @@
+        }
+    }
+
+    @Override
+    public void initErrorMetrics(String serviceType, String type, String metricName, String metricHelp, Map<String,
+            String[]> properties) {
+
+        String[] labels = properties.get(metricName);
+
+        if (serviceType.equals(SERVICE.PROXY.name())) {
+            ERROR_REQUESTS_RECEIVED_PROXY_SERVICE = Counter.build(MetricConstants.PROXY_REQUEST_COUNT_ERROR_TOTAL,
+                    metricHelp).
+                    labelNames(labels).register();
+            metricMap.put(metricName, ERROR_REQUESTS_RECEIVED_PROXY_SERVICE);
+        } else if (serviceType.equals(SERVICE.API.name())) {
+            ERROR_REQUESTS_RECEIVED_API = Counter.build(MetricConstants.API_REQUEST_COUNT_ERROR_TOTAL, metricHelp).
+                    labelNames(labels).register();
+            metricMap.put(metricName, ERROR_REQUESTS_RECEIVED_API);
+        } else if (serviceType.equals(SERVICE.INBOUND_ENDPOINT.name())) {
+            ERROR_REQUESTS_RECEIVED_INBOUND_ENDPOINT = Counter.
+                    build(MetricConstants.INBOUND_ENDPOINT_REQUEST_COUNT_ERROR_TOTAL, metricHelp).labelNames(labels).
+                    register();
+            metricMap.put(metricName, ERROR_REQUESTS_RECEIVED_INBOUND_ENDPOINT);
+
+        }
+    }
+
+    @Override
+    public void incrementCount(String metricName, Map<String, String[]> properties) {
+        Counter counter = (Counter) metricMap.get(metricName);
+        String[] value = properties.get(metricName);
+
+        counter.labels(value).inc();
+    }
+
+    @Override
+    public void decrementCount(String metricName, Map<String, String> properties) {
+
+    }
+
+    @Override
+    public Object getTimer(String metricName, Map<String, String[]> properties) {
+        String[] value = properties.get(metricName);
+        Histogram timer = (Histogram) metricMap.get(metricName);
+
+        return timer.labels(value).startTimer();
+    }
+
+    @Override
+    public void observeTime(Object timer) {
+        Histogram.Timer timer2 = (Histogram.Timer) timer;
+        timer2.observeDuration();
+    }
+
+    @Override
+    public void serverUp(String host, String port, String javaVersion, String javaHome) {
+        Gauge gauge = (Gauge) metricMap.get("wso2_integration_server_up");
+        gauge.labels(host, port, javaHome, javaVersion).set(System.currentTimeMillis());
+    }
+
+    @Override
+    public void serverDown(String host, String port, String javaVersion, String javaHome) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVER_UP);
+        gauge.labels(host, port, javaHome, javaVersion).set(0);
+    }
+
+    @Override
+    public void serviceUp(String serviceName, String serviceType) {
+        Gauge gauge = (Gauge) metricMap.get(MetricConstants.SERVICE_UP);
+        gauge.labels(serviceName, serviceType).set(System.currentTimeMillis());
+        System.out.println(gauge.labels(serviceName, serviceType).get());
```

Comment:
Shall we remove these print statements here, and have logs if needed.

Reply:
Fixed